api/pkg/auth/service/service.go

```diff
@@ -92,57 +91,121 @@ func createChecksum(token string) string {
 	return hex.EncodeToString(hash[:])
 }
 
-func (r *request) insertData(ghUser goth.User, code string) error {
+func (r *request) insertData(gitUser goth.User, code, provider string) error {
```

Comment:
we should document what is going to happen with user data 
while login from multiple provider

Reply:
could you break down this func in multiple smaller func 
depending on the scenario
it is a bit difficult to understand bcz of multiple if-else